### ISSUE TITLE
Kubernetes config with 'exec' resource where exec requires MFA hangs

### ISSUE THREAD

**Issue by MikaelSmith** [puppetlabs/wash#297]
A common way to configure AWS EKS in kube config is
```
user:
  exec:
    apiVersion: client.authentication.k8s.io/v1alpha1
    command: aws-iam-authenticator
    args: [token, -i, infracore-test-1, --cache]
    env:
    - name: AWS_PROFILE
      value: <profile>
```

This will cache and re-use an MFA token if using an account with MFA, but if the token needs to be re-requested Wash hangs, then eventually times out and crashes the terminal.

We need to identify and expose that request for an MFA token when it happens.

----

For completeness, the way the AWS CLI tool sets up kube config is with
```
user:
  exec:
    apiVersion: client.authentication.k8s.io/v1alpha1
    command: aws
    args: [eks get-token --cluster-name <cluster>]
    env:
    - name: AWS_PROFILE
      value: <profile>
```
which doesn't cache the token at all, so it attempts to request an MFA token every time you run kubectl.

**Comment by MikaelSmith** [comment]
Fix for this conflicted with some tools. When reading a file or directory with MacVim, I get
```
zsh: error on TTY read: Input/output error
```
and Wash tries to exit.

**Comment by MikaelSmith** [comment]
The challenge on this is that the Wash daemon is the top of the process group that handles plugin invocation. If a plugin needs to prompt for input, the Wash daemon needs to have control of the terminal to read stdin or the shell will error when Wash prompts for stdin.

The child shell that Wash starts is in its own process group. Trying to read input while another process group has control of stdin results in an error. So the first thing to look at is whether we can get Wash and its child shell in the same process group. If not, stealing control temporarily mostly works but occasionally causes errors if a command is zsh attempts a new read on stdin while we're doing it (or possibly something else with MacVim).

**Comment by MikaelSmith** [comment]
These kind of prompts also interact poorly with the plan for #302 to have the daemon run in a separate process.

**Comment by MikaelSmith** [comment]
https://blog.nelhage.com/2011/02/changing-ctty/ gave me a much simpler idea for this. We make the Wash daemon part of the shell process it started.

**Comment by MikaelSmith** [comment]
Fixed by #635.
